core/storage/trie/polkadot_trie_db/polkadot_codec.cpp

```diff
@@ -43,78 +43,64 @@ namespace kagome::storage::trie {
 
   inline common::Buffer ushortToBytes(uint16_t b) {
     common::Buffer out(2, 0);
-    out[0] = (b >> 8u) & 0xffu;
-    out[1] = b & 0xffu;
+    out[1] = (b >> 8u) & 0xffu;
+    out[0] = b & 0xffu;
     return out;
   }
 
   common::Buffer PolkadotCodec::nibblesToKey(const common::Buffer &nibbles) {
-    const auto nibbles_size = nibbles.size();
-    if (nibbles_size == 0) {
-      return {};
-    }
-
-    if (nibbles_size == 1 && nibbles[0] == 0) {
-      return {0};
-    }
-
-    // if nibbles_size is odd, then allocate one more item
-    const size_t size = (nibbles_size + 1) / 2;
-    common::Buffer out(size, 0);
-
-    size_t iterations = size;
-
-    // if number of nibbles is odd, then iterate even number of times
-    bool nimbles_size_odd = nibbles_size % 2 != 0;
-    if (nimbles_size_odd) {
-      --iterations;
-    }
-
-    for (size_t i = 0; i < iterations; ++i) {
-      out[i] = collectByte(nibbles[2 * i], nibbles[2 * i + 1]);
-    }
-
-    // if number of nibbles is odd, then implicitly add 0 as very last nibble
-    if (nimbles_size_odd) {
-      out[iterations] = collectByte(nibbles[2 * iterations], 0);
+    Buffer res;
+    if (nibbles.size() % 2 == 0) {
+      res = Buffer(nibbles.size() / 2, 0);
+      for (size_t i = 0; i < nibbles.size(); i += 2) {
+        res[i / 2] = (nibbles[i] << 4 & 0xf0) | (nibbles[i + 1] & 0xf);
```

Comment:
as I recap there is a func `nibblesToKey` or something like that, which does exactly this. maybe use it?